src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2724,6 +2724,15 @@ bool Creature::SetSwim(bool enable)
     return true;
 }
 
+/**
+ * @brief This method check the current flag/status of a creature and its inhabit type
+ * 
+ * Pets should swim by default to properly follow the player
+ * NOTE: You can set the UNIT_FLAG_CANNOT_SWIM temporary to deny a creature to swim
+ * 
+ * @return true 
+ * @return false 
```

Comment:
do we need this? as the return type is bool, it' kind of obvious that it's going to be true or false :D

Reply:
it's automatically generated to allow you documenting what true means and what false means